File: packages/typia/src/programmers/internal/json_schema_string.ts
import { OpenApi } from "@samchon/openapi";

import { MetadataAtomic } from "../../schemas/metadata/MetadataAtomic";

import { json_schema_plugin } from "./json_schema_plugin";

export const json_schema_string = (
  atomic: MetadataAtomic,
): OpenApi.IJsonSchema[] =>
  json_schema_plugin({
    schema: {
      type: "string",
    } satisfies OpenApi.IJsonSchema,
    tags: atomic.tags,
  });

File: packages/typia/native/core/programmers/iterate/json_schema_string.go
package iterate

import nativemetadata "github.com/samchon/typia/packages/typia/native/core/schemas/metadata"

func json_schema_string(atomic *nativemetadata.MetadataAtomic) []JsonSchema {
  return json_schema_plugin(struct {
    schema JsonSchema
    tags   [][]nativemetadata.IMetadataTypeTag
  }{
    schema: JsonSchema{"type": "string"},
    tags:   atomic.Tags,
  })
}

func Json_schema_string_export(atomic *nativemetadata.MetadataAtomic) []JsonSchema {
  return json_schema_string(atomic)
}
